src/main/java/com/example/onlinebookstore/dto/book/CreateBookRequestDto.java

```diff
@@ -20,4 +21,6 @@ public class CreateBookRequestDto {
     private BigDecimal price;
     private String description;
     private String coverImage;
+    @NotNull
+    private Set<Long> categoryIds;

```

Comment:
The set still may be empty. Is it expected?
You can validate this using @NotEmpty.